libr/bin/format/mach0/mach0.c

```diff
@@ -3250,42 +3288,68 @@ struct import_t *MACH0_(get_imports)(struct MACH0_(obj_t) *bin) {
 		return NULL;
 	}
 
-	struct import_t *imports = calloc (bin->dysymtab.nundefsym + 1, sizeof (struct import_t));
-	if (!imports) {
-		return NULL;
-	}
-	for (i = j = 0; i < bin->dysymtab.nundefsym; i++) {
-		idx = bin->dysymtab.iundefsym + i;
+	int i, num_imports;
+	for (i = num_imports = 0; i < bin->dysymtab.nundefsym; i++) {
+		int idx = bin->dysymtab.iundefsym + i;
 		if (idx < 0 || idx >= bin->nsymtab) {
 			R_LOG_WARN ("Imports index out of bounds. Ignoring relocs");
-			free (imports);
 			return NULL;
 		}
-		stridx = bin->symtab[idx].n_strx;
+
+		int stridx = bin->symtab[idx].n_strx;
 		char *imp_name = get_name (bin, stridx, false);
-		if (imp_name) {
-			r_str_ncpy (imports[j].name, imp_name, R_BIN_MACH0_STRING_LENGTH - 1);
-			free (imp_name);
-		} else {
-			//imports[j].name[0] = 0;
+		if (!imp_name) {
 			continue;
 		}
-		imports[j].ord = i;
-		imports[j++].last = 0;
+
+		RBinImport *import = import_from_name (bf->rbin, imp_name);
+		if (!import) {
+			free (imp_name);
+			break;  // TODO change into continue?
+		}
+
+		import->ordinal = i;
+		r_pvector_push (&bin->imports_cache, import);
+
+		num_imports++;
+		if (!strcmp (import->name, "__stack_chk_fail") ) {
```

Comment:
Move into separate function for readability

Reply:
Good point.